core/state_transition.go

```diff
@@ -281,7 +297,11 @@ func (st *StateTransition) TransitionDb() (*ExecutionResult, error) {
 		// After EIP-3529: refunds are capped to gasUsed / 5
 		st.refundGas(params.RefundQuotientEIP3529)
 	}
-	st.state.AddBalance(st.evm.Context.Coinbase, new(big.Int).Mul(new(big.Int).SetUint64(st.gasUsed()), st.gasPrice))
+	effectiveTip := st.gasPrice
```

Comment:
@lightclient shouldn't the `st.gasPrice` already be the effective gas price at this point?

Reply:
Effective gas price is the effective price the user pays (total). Here, we're calculating the effective tip. This is the amount that is paid out to the miner.

Reply:
> Effective gas price is the effective price the user pays (total). Here, we're calculating the effective tip. This is the amount that is paid out to the miner.